src/main/java/uk/gov/companieshouse/itemhandler/ItemHandlerApplication.java

```diff
@@ -2,12 +2,43 @@
 
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
+import uk.gov.companieshouse.environment.EnvironmentReader;
+import uk.gov.companieshouse.environment.exception.EnvironmentVariableException;
+import uk.gov.companieshouse.environment.impl.EnvironmentReaderImpl;
+import uk.gov.companieshouse.itemhandler.environment.EnvironmentVariables;
+import uk.gov.companieshouse.itemhandler.logging.LoggingUtils;
+import uk.gov.companieshouse.logging.Logger;
 
 @SpringBootApplication
 public class ItemHandlerApplication {
 
+    private static EnvironmentReader environmentReader;
+    private static Logger LOGGER = LoggingUtils.getLogger();
+
 	public static void main(String[] args) {
-		SpringApplication.run(ItemHandlerApplication.class, args);
+	    environmentReader = new EnvironmentReaderImpl();
+	    if(checkEnvironmentVariables()) {
```

Comment:
A possibility could be to throw an unchecked exception from the `checkEnvironmentVariables()` method so that no `if` is required?

Reply:
We would have to log the config variable name before checking and if an exception was found the application would exit instead of logging any other missing variables

Reply:
Yes, perhaps this sort of thing would be easier if the `EnvironmentReader` class provided a `checkRequiredEnvironmentVariables` member that checked for the presence of all required variables.